pkg/container-utils/containerd/containerd.go

```diff
@@ -85,25 +80,30 @@ func (c *ContainerdClient) Close() error {
 }
 
 func (c *ContainerdClient) GetContainers() ([]*runtimeclient.ContainerData, error) {
-	containers, err := c.client.Containers(c.ctx)
+	ctx := namespaces.WithNamespace(context.TODO(), getContainerdNamespace())
+	return c.getContainersInNamespace(ctx)
+}
```

Comment:
Could you please add some tests using https://github.com/inspektor-gadget/inspektor-gadget/pull/1850 to verify that a containerd runtime client can retrieve containers in different namespaces? It shouldn't be challenging to modify the [defaultNamespaec](https://github.com/inspektor-gadget/inspektor-gadget/blob/02d21cb3a9c45e1ed1b32ba804d2d606913dd070/pkg/container-utils/testutils/containerd.go#L35) by adding a new [option](https://github.com/inspektor-gadget/inspektor-gadget/blob/02d21cb3a9c45e1ed1b32ba804d2d606913dd070/pkg/container-utils/testutils/options.go#L24) like `WithContainerdNamespace`. WDYT?

BTW, you should manually set the `namespaces.NamespaceEnvVar` in the tests. That's another reason why I'm not convinced about this implementation.